### PR TITLE
added username field like suggested in issue comment

### PR DESCRIPTION
**Pull request by edewit**
related: #32522
Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>

image

### CODE REVIEW

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: themes/src/main/resources/theme/keycloak.v2/login/template.ftl [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+    <div class="${properties.kcInputGroup}">
+      <div class="${properties.kcInputGroupItemClass} ${properties.kcFill}">
+        <span class="${properties.kcInputClass} ${properties.kcFormReadOnlyClass}">
+          <input id="username" value="${auth.attemptedUsername}" readonly>
+        </span>
+      </div>
+      <div class="${properties.kcInputGroupItemClass}">
+        <button tabindex="-1" class="${properties.kcFormPasswordVisibilityButtonClass} kc-login-tooltip" type="button" aria-label="${msg('restartLoginTooltip')}"
+              onclick="location.href='${url.loginRestartFlowUrl}'">
+            <i class="fa-sync-alt fas" aria-hidden="true"></i>
```
[the remark this answers is not in the document]
Comment (edewit) [reply themes/src/main/resources/theme/keycloak.v2/login/template.ftl:22]: This one is still supported in PF5

File: themes/src/main/resources/theme/keycloak.v2/login/template.ftl [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+      </label>
+    </div>
+    <div class="${properties.kcInputGroup}">
+      <div class="${properties.kcInputGroupItemClass} ${properties.kcFill}">
+        <span class="${properties.kcInputClass} ${properties.kcFormReadOnlyClass}">
+          <input id="username" value="${auth.attemptedUsername}" readonly>
+        </span>
+      </div>
+      <div class="${properties.kcInputGroupItemClass}">
+        <button tabindex="-1" class="${properties.kcFormPasswordVisibilityButtonClass} kc-login-tooltip" type="button" aria-label="${msg('restartLoginTooltip')}"
```
[the remark this answers is not in the document]
Comment (edewit) [reply themes/src/main/resources/theme/keycloak.v2/login/template.ftl:20]: this is a very black humor joke, you want people that are blind to be able to "show password"?

[the remark this answers is not in the document]
Comment (ahus1) [reply themes/src/main/resources/theme/keycloak.v2/login/template.ftl:20]: Like comment in the other issue: IMHO it is important to make it navigatable by keyboard for those you can't use a mouse but can see just fine.

On another note: Looking at the label, this is not about showing the password, but about restarting the login. This control is IMHO important for everyone.

**Comment by ahus1** [comment]
Thank you for the visual design, IMHO it is a lot clearer now what's happening here.

**Review by jonkoops — DISMISSED** [review]

**Review by agagancarczyk — APPROVED** [review]

**Review by ahus1 — APPROVED** [review]
Approving based in my manual test and @agagancarczyk's review. Thank you for this enhancement!

**Pull request merged**